sklearn/preprocessing/_encoders.py

```diff
@@ -797,6 +809,14 @@ def fit(self, X, y=None):
         self
             Fitted encoder.
         """
+        if self.sparse != "deprecated":
+            warnings.warn(
+                "'sparse' was renamed to 'sparse_output' in version 1.2 and "
+                "will be removed in 1.4.",
+                FutureWarning,
+            )
+            self.sparse_output = self.sparse
```

Comment:
This is an annoying case here.

Supposedly, we should be raising an error if both `sparse` and `sparse_output` are set to non-default values. However, to detect it we need to add a `None` option to `sparse_output` that defaults to `True` by default. However, we don't want `None` and we also need to depreciate it at the inclusion such that we remove it. It means that in practice we will always be raising a warning and force users to set it to a bool.

@jeremiedbb I recall that we had a similar pattern somewhere else (DictionaryLearning of NMF). How did we do it?

@lorentzenchr @thomasjpfan do you see a better way?

Reply:
To keep it simple, I would raise in the following case
```python
if self.sparse != "deprecated" and self.sparse != self.sparse_output:
    raise ValueError("Some informative message")
```